helios-testing/src/main/java/com/spotify/helios/testing/TemporaryJobJsonReports.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.testing;
+
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.spotify.helios.common.Json;
+import com.spotify.helios.testing.descriptors.TemporaryJobEvent;
+
+import org.junit.runner.Description;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+/**
+ * Reports a stream of TemporaryJobEvent events to a JSON file.
+ */
+public class TemporaryJobJsonReports extends TemporaryJobReports {
+
+  private static final Logger log = LoggerFactory.getLogger(TemporaryJobJsonReports.class);
+
+  private final Path outputDir;
+
+  public TemporaryJobJsonReports(final Path outputDir) {
+    this.outputDir = outputDir;
+    this.outputDir.toFile().mkdirs();
+  }
+
+  public ReportWriter getWriterForTest(final Description testDescription) {
+    return new JsonReportWriter(
+        outputDir, testDescription.getClassName(), testDescription.getMethodName());
+  }
+
+  public ReportWriter getDefaultWriter() {
+    return new JsonReportWriter(System.err);
+  }
+
+  private static class JsonReportWriter extends ReportWriter {
+
+    private final JsonGenerator jg;
+
+    private final String testClassName;
+    private final String testName;
+
+    private JsonReportWriter(
+        final Path outputDir, final String testClassName, final String testName) {
+      this.testClassName = testClassName;
+      this.testName = testName;
+
+      final String logFilename = format("%s.%s.json", testClassName, testName).replace('$', '_');
+      final File logFile = outputDir.resolve(logFilename).toFile();
+
+      JsonGenerator jg = null;
+      try {
+        jg = new JsonFactory().createGenerator(logFile, JsonEncoding.UTF8);
+        jg.writeStartArray();
+      } catch (IOException e) {
+        log.error("exception creating event log: {} - {}", logFile.getAbsolutePath(), e);
+      } finally {
+        this.jg = jg;
```

Comment:
I guess this comes from the original source, but fwiw this pattern with the finally looks strange to me. I think this could just be:

```java
try {
  this.jg = new JsonFactory().createGenerator(logFile, JsonEncoding.UTF8)
} catch (IOException e) {
  //log the exception, and set the final field
  this.jg = null;
}
```

or arguably this field doesn't need to be `final` if it can be null sometimes and set to an instance other times.